core/src/avm1/globals/as_broadcaster.rs

```diff
@@ -0,0 +1,249 @@
+            }
+        }
+    }
+
+    Ok(removed.into())
+}
+
+pub fn broadcast_message<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let event_name_value = args.get(0).cloned().unwrap_or(Value::Undefined);
+    let event_name = event_name_value.coerce_to_string(activation, context)?;
+    let call_args = &args[1..];
+
+    broadcast_internal(activation, context, this, call_args, &event_name)
+}
+
+pub fn broadcast_internal<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    this: Object<'gc>,
+    call_args: &[Value<'gc>],
+    method_name: &str,
+) -> Result<Value<'gc>, Error<'gc>> {
+    let listeners = this.get("_listeners", activation, context)?;
+
+    if let Value::Object(listeners) = listeners {
+        for i in 0..listeners.length() {
+            let listener = listeners.array_element(i);
+
+            if let Value::Object(listener) = listener {
+                listener.call_method(method_name, call_args, activation, context)?;
+            }
+        }
+    }
+
+    Ok(Value::Undefined)
+}
+
+pub fn initialize<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    _this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    if let Some(val) = args.get(0) {
+        let broadcaster = val.coerce_to_object(activation, context);
+        initialize_internal(
+            context.gc_context,
+            broadcaster,
+            activation.avm.broadcaster_functions,
+            activation.avm.prototypes().array,
+        );
+    }
+    Ok(Value::Undefined)
+}
+
+pub fn initialize_internal<'gc>(
+    gc_context: MutationContext<'gc, '_>,
+    broadcaster: Object<'gc>,
+    functions: BroadcasterFunctions<'gc>,
+    array_proto: Object<'gc>,
+) {
+    let listeners = ScriptObject::array(gc_context, Some(array_proto));
+
+    broadcaster.define_value(gc_context, "_listeners", listeners.into(), DontEnum.into());
+
+    broadcaster.define_value(
+        gc_context,
+        "addListener",
+        functions.add_listener.into(),
+        DontEnum.into(),
+    );
+
+    broadcaster.define_value(
+        gc_context,
+        "removeListener",
+        functions.remove_listener.into(),
+        DontEnum.into(),
+    );
+
+    broadcaster.define_value(
+        gc_context,
+        "broadcastMessage",
+        functions.broadcast_message.into(),
+        DontEnum.into(),
+    );
+}
+
+pub fn create<'gc>(
+    gc_context: MutationContext<'gc, '_>,
+    proto: Option<Object<'gc>>,
+    fn_proto: Option<Object<'gc>>,
+) -> (BroadcasterFunctions<'gc>, Object<'gc>) {
+    let mut as_broadcaster = ScriptObject::object(gc_context, proto);
+
+    as_broadcaster.force_set_function(
+        "initialize",
+        initialize,
+        gc_context,
+        DontDelete | ReadOnly | DontEnum,
```

Comment:
Seems like all the properties are only `DontDelete | DontEnum` -- you can overwrite them.
```
trace(_global.AsBroadcaster.removeListener);
_global.AsBroadcaster.removeListener = 1;
delete _global.AsBroadcaster.removeListener;
trace(_global.AsBroadcaster.removeListener);
```